### DISCUSSION TITLE
[Auxkernel][Converge]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#21055] [Q&A General] [answered]
Dear Moose users,

I want to ask a tiny stupid question.

As far as I know, auxkernel and its auxvariable just calculating some variables.

And the auxvariable and auxkernel should be nothing related to a convergence of a numerical calculation.

Am I right?

I am a little bit confused now because I added some auxvariables. And it contributes to the convergence of the entire calculation.

I didn't make any coupling from these auxvariables to other kernels or material properties.

To me, this is so weird.....

**Comment by GiudGiud** [comment, marked as answer]
Hello

It's not necessarily unrelated to the numerical calculation. The auxvariables may be used in the numerical scheme to compute material properties for example. This all depends on how you write the problem in. 
For example:
T is your main variable
equation : dT/dt + D(T) grad_T = 0

if you make D a material property you get an updated value throughout the nonlinear solve (by default)
if you make D an auxvariable you get an updated value whenever you execute the auxkernel for D. If you execute it on TIMESTEP_END for example, you will consider D to be constant on each time step, so basically your numerical scheme will lag D. 
But the default for auxkernel is actually (for most) linear + timestep_end. So you will get updates the same way as a material property. 

Guillaume

**Reply by coskrrb2002** [reply]
Thank you for your advice!

I found the error...
I used the wrong mesh....

Thank you!
